### PR TITLE
add upload service

### PR DESCRIPTION
**Pull request by kepta**
Adds uploads service.

### CODE REVIEW

File: services/uploads.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  });
+};
+
+/**
+ * Create an upload.
+ *
+ * See the [public documentation](https://www.mapbox.com/api-documentation/#create-an-upload).
+ *
+ * @param {Object} [config
+ * @param {string} config.tileset - the map ID to create or replace in the format  username.nameoftileset - limited to 32 characters (only  - and  _ special characters allowed, limit does not include username)
```
Comment (davidtheclark) [inline services/uploads.js]: Let's call this `mapId` — I think that pretty consistently should refer to the `{ownerId}.{tilesetId}` pairing, whereas `tileset` is kind of ambiguous.

File: services/uploads.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ * @param {string} config.url - HTTPS URL of the S3 object provided in the credential request or the dataset ID of an existing Mapbox dataset to be uploaded
```
Comment (davidtheclark) [inline services/uploads.js]: How about `s3Url`, to make it clear that this is not the URL of the request you're making?

File: services/uploads.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+ * @param {string} [config.name] - name for the tileset - limited to 64 characters
```
Comment (davidtheclark) [inline services/uploads.js]: How about `tilesetName`, so it's clear you're naming the tileset not the upload?

File: services/uploads.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    path: '/uploads/v1/:ownerId/credentials'
+  });
+};
+
+/**
+ * Create an upload.
+ *
+ * See the [public documentation](https://www.mapbox.com/api-documentation/#create-an-upload).
+ *
+ * @param {Object} [config
```
Comment (davidtheclark) [inline services/uploads.js]: Looks like there stray `[`s here and below.

**Comment by kepta** [comment]
Yup I tried it out, but I used the dataset instead of s3. Which I feel should be the same ?

**Comment by davidtheclark** [comment]
👍

**Pull request merged**